lint.go

```diff
@@ -1038,22 +1038,38 @@ func (f *file) lintRanges() {
 	})
 }
 
-// lintErrorf examines errors.New calls. It complains if its only argument is an fmt.Sprintf invocation.
+// lintErrorf examines errors.New and testing.Error calls. It complains if their only arguments are fmt.Sprintf invocations.
 func (f *file) lintErrorf() {
+	testingTName := ""
 	f.walk(func(node ast.Node) bool {
-		ce, ok := node.(*ast.CallExpr)
-		if !ok {
-			return true
-		}
-		if !isPkgDot(ce.Fun, "errors", "New") || len(ce.Args) != 1 {
-			return true
-		}
-		arg := ce.Args[0]
-		ce, ok = arg.(*ast.CallExpr)
-		if !ok || !isPkgDot(ce.Fun, "fmt", "Sprintf") {
-			return true
+		switch ce := node.(type) {
+		case *ast.FuncDecl:
+			name := ce.Name.Name
+			if strings.HasPrefix(name, "Test") {
```

Comment:
Let's lean on the typechecker a bit more here. Instead of tracking whether we're in a test func, use the typechecker when you see `t.Error` to identify the type of `t`. Use `f.pkg.typeOf` and `f.pkg.isNamedType` (after checking for types.Pointer). Then this walk only needs to inspect `CallExpr`s again.